packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -930,12 +930,20 @@ export class ContextualMenuBase extends BaseComponent<IContextualMenuProps, ICon
       this._executeItemClick(item, ev);
     } else {
       if (item.key !== this.state.expandedMenuItemKey) {
+        let expandedByMouseClick = false; // keyboard event
+        if (ev.type === 'click') {
+          // When Edge + Narrator are used together (regardless of if the button is in a form or not), pressing
+          // "Enter" fires this method and not _onMenuKeyDown. Checking ev.nativeEvent.screenX and ev.
+          // nativeEvent.screenY differentiates between a real click event and a keypress event.
+          const clickEv = ev as React.MouseEvent<any>;
+          expandedByMouseClick = clickEv.nativeEvent.screenX !== 0 && clickEv.nativeEvent.screenY !== 0;
```

Comment:
This seems a bit hacky, is there something else we could do here? (or scope this to only run in IE?)

Reply:
At the very least we should have an isMouseEvent function somewhere common so we can keep them in sync.

Reply:
You could try to use Event.type or this might be a valid case for instanceof as well. You can use instance of by doing something like `clickEv.nativeEvent instanceof MouseEvent`

Reply:
@jspurlin Agreed, but I did a comparison, and the only thing that was different besides the detail were the screen size (and other x,y position values).

I think the only way we can make this better is that we actually have 2 separate functions. One for the click event and one for when the user opens it with a keyboard event. However this will require some refactoring.

@joschect I'll add a common function somewhere, that definitely makes sense to do that this point.

Regarding your 2nd comment, do you mean that I should cast the event that way: clickEv.nativeEvent instanceof MouseEvent

Or I should try to use that to tell the difference between the click vs keyboard event.

If it's the 2nd point, unfortunately that doesn't work, this is an actual click event that's generated because the enter keyboard event wasn't handled.